--- app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryViewModel.kt
+++ app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryViewModel.kt
@@ -282,14 +282,12 @@
                             val newDownloadCount = downloadCountMap[item.displayManga.mangaId] ?: 0
                             val newTrackCount = trackMap[item.displayManga.mangaId]?.size ?: 0

                             if (item.downloadCount == newDownloadCount && item.trackCount == newTrackCount) {
-                                item
-                            } else {
-                                item.copy(
-                                    downloadCount = newDownloadCount,
-                                    trackCount = newTrackCount,
-                                )
-                            }
+                                return@map item
+                            }
+                            item.copy(
+                                downloadCount = newDownloadCount,
+                                trackCount = newTrackCount,
+                            )
                         }
                         .applyFilters(libraryFilters, trackMap)
                 }
