--- app/src/main/java/eu/kanade/tachiyomi/AppModule.kt
+++ app/src/main/java/eu/kanade/tachiyomi/AppModule.kt
@@ -2,7 +2,6 @@

 import android.app.Application
-import androidx.core.content.ContextCompat
 import androidx.work.WorkManager
 import eu.kanade.tachiyomi.data.cache.ChapterCache
@@ -53,6 +52,9 @@
 import tachiyomi.core.preference.PreferenceStore
 import tachiyomi.core.util.storage.AndroidStorageFolderProvider
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 import uy.kohesive.injekt.api.InjektModule
 import uy.kohesive.injekt.api.InjektRegistrar
@@ -169,8 +171,8 @@
         addSingletonFactory { MangaShortcutManager() }

         // Asynchronously init expensive components for a faster cold start
-        ContextCompat.getMainExecutor(app).execute {
+        CoroutineScope(Dispatchers.Default).launch {
             get<NetworkHelper>()

             get<SourceManager>()
