--- core/src/main/kotlin/tachiyomi/core/util/storage/DiskUtil.kt
+++ core/src/main/kotlin/tachiyomi/core/util/storage/DiskUtil.kt
@@ -5,6 +5,7 @@
 import android.net.Uri
 import android.os.Environment
 import android.os.StatFs
+import android.media.MediaScannerConnection
 import android.text.format.Formatter
 import androidx.core.content.ContextCompat
 import com.hippo.unifile.UniFile
@@ -87,14 +88,14 @@

     /** Scans the given file so that it can be shown in gallery apps, for example. */
     fun scanMedia(context: Context, file: File) {
-        scanMedia(context, Uri.fromFile(file))
+        MediaScannerConnection.scanFile(context, arrayOf(file.absolutePath), null, null)
     }

     /** Scans the given file so that it can be shown in gallery apps, for example. */
     fun scanMedia(context: Context, uri: Uri) {
-        val action = Intent.ACTION_MEDIA_SCANNER_SCAN_FILE
-        val mediaScanIntent = Intent(action)
-        mediaScanIntent.data = uri
-        context.sendBroadcast(mediaScanIntent)
+        val path = uri.path
+        if (path != null) {
+            MediaScannerConnection.scanFile(context, arrayOf(path), null, null)
+        }
     }
