--- ./app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaViewModel.kt
+++ ./app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaViewModel.kt
@@ -285,15 +285,13 @@
                 mangaDexPreferences.blockedUploaders().changes(),
             ) { dbChapters, blockedGroups, blockedUploaders ->
                 val dbManga = db.getManga(mangaId).executeAsBlocking()!!
-                dbChapters
-                    .mapNotNull { it.toSimpleChapter() }
-                    .filter { chapter ->
+                dbChapters.mapNotNull { dbChapter ->
+                    dbChapter.toSimpleChapter()?.takeIf { chapter ->
                         val scanlators = chapter.scanlatorList()
                         scanlators.none { scanlator -> blockedGroups.contains(scanlator) } &&
                             (Constants.NO_GROUP !in scanlators ||
                                 chapter.uploader !in blockedUploaders)
-                    }
-                    .map { chapter ->
+                    }?.let { chapter ->
                         val downloadState =
                             when {
                                 downloadManager.isChapterDownloaded(
@@ -323,6 +321,7 @@
                                 },
                         )
                     }
+                }
             }
             .distinctUntilChanged()
             .shareIn(viewModelScope, SharingStarted.WhileSubscribed(5000), 1)
