--- app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryViewModel.kt
+++ app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryViewModel.kt
@@ -557,98 +557,51 @@
     }

     fun preferenceUpdates() {
-        preferences
-            .useVividColorHeaders()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(useVividColorHeaders = enabled) }
-            }
-            .launchIn(viewModelScope)
+        fun <T> Flow<T>.observeAndUpdate(update: (LibraryScreenState, T) -> LibraryScreenState) {
+            this.distinctUntilChanged()
+                .onEach { value -> _internalLibraryScreenState.update { state -> update(state, value) } }
+                .launchIn(viewModelScope)
+        }

-        libraryPreferences
-            .showStartReadingButton()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(showStartReadingButton = enabled) }
-            }
-            .launchIn(viewModelScope)
+        preferences.useVividColorHeaders().changes().observeAndUpdate { state, value ->
+            state.copy(useVividColorHeaders = value)
+        }

-        mangadexPreferences
-            .includeUnavailableChapters()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(showUnavailableFilter = enabled) }
-            }
-            .launchIn(viewModelScope)
+        libraryPreferences.showStartReadingButton().changes().observeAndUpdate { state, value ->
+            state.copy(showStartReadingButton = value)
+        }

-        securityPreferences
-            .incognitoMode()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(incognitoMode = enabled) }
-            }
-            .launchIn(viewModelScope)
+        mangadexPreferences.includeUnavailableChapters().changes().observeAndUpdate { state, value ->
+            state.copy(showUnavailableFilter = value)
+        }

-        libraryPreferences
-            .outlineOnCovers()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(outlineCovers = enabled) }
-            }
-            .launchIn(viewModelScope)
+        securityPreferences.incognitoMode().changes().observeAndUpdate { state, value ->
+            state.copy(incognitoMode = value)
+        }

-        libraryPreferences
-            .showDownloadBadge()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(showDownloadBadges = enabled) }
-            }
-            .launchIn(viewModelScope)
+        libraryPreferences.outlineOnCovers().changes().observeAndUpdate { state, value ->
+            state.copy(outlineCovers = value)
+        }

-        libraryPreferences
-            .showUnreadBadge()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(showUnreadBadges = enabled) }
-            }
-            .launchIn(viewModelScope)
+        libraryPreferences.showDownloadBadge().changes().observeAndUpdate { state, value ->
+            state.copy(showDownloadBadges = value)
+        }

-        libraryPreferences
-            .libraryHorizontalCategories()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(horizontalCategories = enabled) }
-            }
-            .launchIn(viewModelScope)
+        libraryPreferences.showUnreadBadge().changes().observeAndUpdate { state, value ->
+            state.copy(showUnreadBadges = value)
+        }

-        libraryPreferences
-            .showLibraryButtonBar()
-            .changes()
-            .distinctUntilChanged()
-            .onEach { enabled ->
-                _internalLibraryScreenState.update { it.copy(showLibraryButtonBar = enabled) }
-            }
-            .launchIn(viewModelScope)
+        libraryPreferences.libraryHorizontalCategories().changes().observeAndUpdate { state, value ->
+            state.copy(horizontalCategories = value)
+        }

-        combine(libraryPreferences.gridSize().changes(), libraryPreferences.layout().changes()) {
-                gridSize,
-                layout ->
-                gridSize to layout
-            }
-            .distinctUntilChanged()
-            .onEach { pair ->
-                _internalLibraryScreenState.update {
-                    it.copy(libraryDisplayMode = pair.second, rawColumnCount = pair.first)
-                }
-            }
-            .launchIn(viewModelScope)
+        libraryPreferences.showLibraryButtonBar().changes().observeAndUpdate { state, value ->
+            state.copy(showLibraryButtonBar = value)
+        }
+
+        combine(libraryPreferences.gridSize().changes(), libraryPreferences.layout().changes()) { gridSize, layout ->
+            gridSize to layout
+        }.observeAndUpdate { state, value ->
+            state.copy(libraryDisplayMode = value.second, rawColumnCount = value.first)
+        }
     }
