--- app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaViewModel.kt
+++ app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaViewModel.kt
@@ -224,8 +224,7 @@
     val mangaFlow =
         db.getManga(mangaId)
-            .asRxObservable()
             .asFlow()
             .map { it.toMangaItem() }
             .distinctUntilChanged()
             .shareIn(viewModelScope, SharingStarted.WhileSubscribed(5000), 1)
