--- app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryViewModel.kt
+++ app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryViewModel.kt
@@ -291,15 +291,14 @@
                         val newTrackCount = trackMap[item.displayManga.mangaId]?.size ?: 0

                         val updatedItem =
-                            if (
-                                item.downloadCount == newDownloadCount &&
-                                    item.trackCount == newTrackCount
-                            ) {
-                                item
-                            } else {
+                            if (item.downloadCount != newDownloadCount || item.trackCount != newTrackCount) {
                                 item.copy(
                                     downloadCount = newDownloadCount,
                                     trackCount = newTrackCount,
                                 )
+                            } else {
+                                item
                             }
-                        if (updatedItem.matchesFilters(libraryFilters, trackMap)) updatedItem
-                        else null
+
+                        updatedItem.takeIf { it.matchesFilters(libraryFilters, trackMap) }
                     }
                 }
             }
@@ -599,15 +598,5 @@
         if (!libraryFilters.filterMerged.matches(this)) return false
         if (!libraryFilters.filterUnavailable.matches(this)) return false
         if (!libraryFilters.filterTracked.matches(this)) return false
-
-        // Special handling for Tracking logic preserved from original code
-        val displayManga = this.displayManga
-        val missingChaptersCondition =
-            when (libraryFilters.filterTracked) {
-                FilterTracked.Inactive -> true
-                FilterTracked.NotTracked -> trackMap[displayManga.mangaId] == null
-                FilterTracked.Tracked -> trackMap[displayManga.mangaId] != null
-            }
-
-        if (!missingChaptersCondition) return false

         return true // passed all checks
     }
